pkg/chunk/azure/blob_storage_client.go

```diff
@@ -77,14 +77,22 @@ func (b *BlobStorage) getChunk(ctx context.Context, decodeContext *chunk.DecodeC
 		return chunk.Chunk{}, err
 	}
 
-	buf := make([]byte, 0, b.cfg.DownloadBufferSize)
+	// Request access to the blob
+	downloadResponse, err := blockBlobURL.Download(ctx, 0, azblob.CountToEnd, azblob.BlobAccessConditions{}, false)
+	if err != nil {
+		return chunk.Chunk{}, err
+	}
+
+	// download the contents & read into a buffer
+	bodyStream := downloadResponse.Body(azblob.RetryReaderOptions{MaxRetryRequests: b.cfg.MaxRetries})
+	buf := bytes.Buffer{}
```

Comment:
I don't think I saw this usage before, maybe just use `bytes.NewBuffer(nil)`?

Reply:
If all we need is to read the entire response in a `[]byte`, then I believe `ioutil.ReadAll(bodyStream)` is preferable.

Reply:
I agree with @pracucci ... ReadAll makes this a bit more tidy to read plus matches all we need to do here.